### PR TITLE
define the service client model

### PR DESCRIPTION
**Pull request by fredbelotte**

### CODE REVIEW

File: src/Housing.Forecast.Api.Client/Controllers/BatchController.cs [hunk 1]
```diff
@@ -2,40 +2,76 @@
...
+      {
+        _context.Batch.Update(value);
+        await _context.SaveChangesAsync();
+      });
     }
 
-    [HttpDelete("{batchId}")]
-    public void Delete(int batchId)
+    [HttpDelete("{BatchId}")]
+    public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/BatchController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1pexh_eEZPtUmVDqP&open=AWW1pexh_eEZPtUmVDqP)

[ISSUE_KEY]: AWW1pexh_eEZPtUmVDqP

File: src/Housing.Forecast.Api.Client/Controllers/TechnologyController.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    {
+      await Task.Run(async () =>
+      {
+        _context.Technology.Update(value);
+        await _context.SaveChangesAsync();
+      });
+    }
+
+    [HttpDelete("{Technologyid}")]
+     public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/TechnologyController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1peyV_eEZPtUmVDqW&open=AWW1peyV_eEZPtUmVDqW)

[ISSUE_KEY]: AWW1peyV_eEZPtUmVDqW

File: src/Housing.Forecast.Api.Client/Controllers/LocationController.cs [hunk 3]
```diff
@@ -2,40 +2,77 @@
...
+      {
+        _context.Location.Update(value);
+        await _context.SaveChangesAsync();
+      });
     }
 
-    [HttpDelete("{locationId}")]
-    public void Delete(int locationId)
+    [HttpDelete("{LocationId}")]
+    public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/LocationController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1peyC_eEZPtUmVDqT&open=AWW1peyC_eEZPtUmVDqT)

[ISSUE_KEY]: AWW1peyC_eEZPtUmVDqT

File: src/Housing.Forecast.Api.Client/Controllers/NameController.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    {
+      await Task.Run(async () =>
+      {
+        _context.Name.Update(value);
+        await _context.SaveChangesAsync();
+      });
+    }
+
+    [HttpDelete("{NameId}")]
+     public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/NameController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1peyF_eEZPtUmVDqU&open=AWW1peyF_eEZPtUmVDqU)

[ISSUE_KEY]: AWW1peyF_eEZPtUmVDqU

File: src/Housing.Forecast.Api.Client/Controllers/BedController.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    {
+      await Task.Run(async () =>
+      {
+        _context.Bed.Update(value);
+        await _context.SaveChangesAsync();
+      });
+    }
+
+    [HttpDelete("{BedId}")]
+    public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/BedController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1pexm_eEZPtUmVDqQ&open=AWW1pexm_eEZPtUmVDqQ)

[ISSUE_KEY]: AWW1pexm_eEZPtUmVDqQ

File: src/Housing.Forecast.Api.Client/Controllers/CampusController.cs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    {
+      await Task.Run(async () =>
+      {
+        _context.Campus.Update(value);
+        await _context.SaveChangesAsync();
+      });
+    }
+
+    [HttpDelete("{CampusId}")]
+     public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/CampusController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1pexy_eEZPtUmVDqR&open=AWW1pexy_eEZPtUmVDqR)

[ISSUE_KEY]: AWW1pexy_eEZPtUmVDqR

File: src/Housing.Forecast.Api.Client/Controllers/UserController.cs [hunk 7]
```diff
@@ -2,40 +2,77 @@
...
+      {
+        _context.User.Update(value);
+        await _context.SaveChangesAsync();
+      });
     }
 
-    [HttpDelete("{userId}")]
-    public void Delete(int userId)
+    [HttpDelete("{UserId}")]
+   public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/UserController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1peyl_eEZPtUmVDqX&open=AWW1peyl_eEZPtUmVDqX)

[ISSUE_KEY]: AWW1peyl_eEZPtUmVDqX

File: src/Housing.Forecast.Api.Client/Controllers/CarController.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    {
+      await Task.Run(async () =>
+      {
+        _context.Car.Update(value);
+        await _context.SaveChangesAsync();
+      });
+    }
+
+    [HttpDelete("{CarId}")]
+    public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/CarController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1pex1_eEZPtUmVDqS&open=AWW1pex1_eEZPtUmVDqS)

[ISSUE_KEY]: AWW1pex1_eEZPtUmVDqS

File: src/Housing.Forecast.Api.Client/Controllers/RoomController.cs [hunk 9]
```diff
@@ -2,40 +2,75 @@
...
+      {
+        _context.Room.Update(value);
+        await _context.SaveChangesAsync();
+      });
     }
 
-    [HttpDelete("{roomId}")]
-    public void Delete(int roomId)
+    [HttpDelete("{RoomId}")]
+   public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/RoomController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1peyS_eEZPtUmVDqV&open=AWW1peyS_eEZPtUmVDqV)

[ISSUE_KEY]: AWW1peyS_eEZPtUmVDqV

File: src/Housing.Forecast.Api.Client/Controllers/AddressController.cs [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    {
+      await Task.Run(async () =>
+      {
+        _context.Address.Update(value);
+        await _context.SaveChangesAsync();
+      });
+    }
+
+    [HttpDelete("{BatchId}")]
+    public async void DeleteAsync(int id)
```
Comment (fredbelotte) [inline src/Housing.Forecast.Api.Client/Controllers/AddressController.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aforecast%3Aapi&pullRequest=8&issues=AWW1pexN_eEZPtUmVDqO&open=AWW1pexN_eEZPtUmVDqO)

[ISSUE_KEY]: AWW1pexN_eEZPtUmVDqO

**Comment by vincenttso** [comment]
the latest two commits are to get rid of build-level errors and to prepare for database integration (moving from mock to forecast context). the ids will not be indicative of actual properties on the object model.

**Pull request merged**